### PR TITLE
Adds in the product client to use the product endpoints for pipedrive.

### LINKED ISSUE #15: Product endpoints not supported
It appears the current implementation does not support the Product endpoints (and related Deal-Product endpoints).
We will be looking to create a pull request in the coming weeks to rectify this; will you be available to look at it? We would very much appreciate it!

Thanks,
Laurence

### PR DESCRIPTION
**Pull request by jbiesta**
Closes #15

### CODE REVIEW

**Comment by jbiesta** [comment]
Hi @DavidRouyer As discussed in #15 here is the code for adding the products endpoint support in.
Please could you have a look when you have some time?
I know you recently had a discussion about data not being created correctly when running the tests (we had this same issue here with some of the tests) so i wasn't entirely sure how/where our data for these tests should be created.
On our pipedrive i added the following data in:
2 products - 1 with a name of productname and a price of 12.32
Requires one follower of the product with id 2
Requires one deal with the product with id 2
Requires one file associated with product with id 2

Please let me know if i need to do this data creation differently (and i'll update the test accordingly)

Thanks

**Comment by DavidRouyer** [comment]
Thank you @jbiesta for your contribution. Can you start by fixing the failing unit tests?

**Comment by jbiesta** [comment]
@DavidRouyer Apologies they should be sorted now :)

File: src/Pipedrive.net/Models/Request/ProductFilters.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                {
+                    d.Add("first_char", FirstCharacter);
+                }
+                return d;
+            }
+        }
+
+    }
+
+    public abstract class BaseFilter
```
Comment (jbiesta) [inline src/Pipedrive.net/Models/Request/ProductFilters.cs]: Remove

Comment (jbiesta) [reply src/Pipedrive.net/Models/Request/ProductFilters.cs]: This should be ApiOptions instead.

**Comment by woodylaurence** [comment]
@DavidRouyer I have added a few small tweaks to this work, adding custom fields to the newly created Product (and related) object.

I also noticed that on the New.. models (NewDeal, NewOrganization etc) we were missing the the CustomFieldConverter attribute, meaning we were unable to create an object with its custom fields set.

I've added a bit of test logic to the DealsClientTests.TheCreateMethod.CanCreate test to check we can set custom fields on create. I've only added it to this, rather than adding it to the tests for Organization, Person etc since these don't have any tests which check custom fields. Let me know if you want this changed.

Thanks,
Laurence

**Comment by redflowIT** [comment]
Hi all
I saw you were working on Products client. Have you any release date?
Thank you

**Comment by DavidRouyer** [comment]
Pipedrive.net v0.3.0 has been released

**Pull request merged**